packages/react/src/data-table/DataTable.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  type ColumnDef,
+  type SortingState,
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import { createElement, useState } from "react";
+
+import { Box } from "../box";
+import { DataTableHeader } from "../data-table-header";
+import { Pagination } from "../pagination";
+import { Table } from "../table";
+import { TableBody } from "../table-body";
+import { TableCell } from "../table-cell";
+import { TableHead } from "../table-head";
+import { TableHeaderCell } from "../table-header-cell";
+import { TableRow } from "../table-row";
+import * as styles from "./DataTable.css";
+
+interface DataTableProps<TData, TValue> {
+  columns: ColumnDef<TData, TValue>[];
+  data: TData[];
+  pinnedColumns?: string[];
+}
```

Comment:
this doesn't look right - we should re-use the tanstack Table type - something like:
```suggestion
import type { TableOptions } from "@tanstack/react-table";

type DataTableProps<TData> = TableOptions<TData>;

export const DataTable = <TData,>({ columns, data, ...props }: DataTableProps<TData>) => {
```